torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,32 +65,95 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: torch.dtype
```

Comment:
if this dtype can be None, then we don't need Optional[DTypeWithConstraints]

Reply:
Ok, I see what you mean. I'll make the change